Multiifo config dev mk2

tapaimarton opened the pull request:
These are the config file changes required for https://github.com/gwastro/pycbc/pull/2383 to work.

Commit:
Adding c02 data ini file for O3 multi ifo dev tests

Commit:
Added correct data.ini and data_C02.ini for multi ifo configuration

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Adding changes required for multiifo workflow

tdent commented:
@tapaimarton it will quickly become confusing if I am making comments on two requests at two different repos at once. please tell me how to proceed

tapaimarton commented:
I think we should move the discussion here, so @ahnitz can comment on it as well. I can apply the changes to the other repo once we are all ok with it here.

Commit:
Adding commit about strict coincidence

tdent commented on O3exp/pipeline/analysis.ini:
```diff
@@ -71,8 +71,27 @@ do-trigger-fitting =
+
+[workflow-coincidence-full2d0]
+; calculates H-L coincidences over all times when H1 and L1 are observing
+parallelization-factor = 1
```
btw, why is parallelization-factor changed from 20 to 1 here, is the new code much more efficient or does it not yet handle parallelization?

tapaimarton replied:
It handles it, I just wanted to create 1 file for each case while testing. I can set that back to 20.

tdent replied:
ok, but I think we should keep 20 as a standard value unless @ahnitz disagrees or there is a definite reason to change.

tdent commented on O3exp/pipeline/analysis.ini:
```diff
@@ -236,27 +255,45 @@ smoothing-width = 0.2
+; time-slide interval in seconds
+timeslide-interval = 0.1
+; reduction factor to make lightning bolts for IFAR plots
+decimation-factor = 5000
```
do we need a high decimation factor and/or loudest keep value for triples, in fact?

tapaimarton replied:
Not necessarily, I kept everything at the O2 values for these, my aim was to get the code to work first.

tdent commented on O3exp/pipeline/plotting.ini:
```diff
@@ -427,7 +427,7 @@ bin-spacing = log
 
 [page_ifar]
-decimation-factor = ${coinc-full|decimation-factor}
+decimation-factor = ${multiifo_coinc-full3d|decimation-factor}
```
note that we might not necessarily have the same decimation factor for all coinc jobs.  
(& I wonder do we really need to give the ifar plot code the decimation factor as an option .. ie prefer it to be stored in the coinc trigger files?)

tapaimarton replied:
I think the decimation factor is stored in the coinc trigger files, (was stored in the old version as well)

Commit:
changes for new workflow generation version

tdent commented on O3exp/pipeline/analysis.ini:
```diff
@@ -72,6 +72,7 @@ do-trigger-fitting =
 ;background-bins = all:total:lt600
 
 [workflow-coincidence-full]
+timeslide-precedence = H1, L1, V1, K1, I1
```
can you propose a comment line here to describe what this option does?

tapaimarton replied:
It took more than a line, but I think this description should be clear enough, is it ok?

tdent replied:
@ahnitz @tapaimarton can I check what this actually means, without the 'pivot'/'fixed' jargon : out of any given combination of detectors, with the current implementation *only one* ifo is slid relative to all the others, and that ifo is called 'pivot' ? is that correct?

ahnitz replied:
@tdent Yes, you can think of it that way.

tdent replied:
OK, would there be any objection to documenting it in the ini file simply as this : 

"for a given ifo combination, time shifted coincidences are generated by shifting one ifo relative to all the rest: that ifo is chosen to be the one appearing first in this list"

The user doesn't need to know everything that goes on under the hood given that only the choice of 'pivot' makes a difference at present.  If/when the 'pivot' and 'fixed' designations mean something more specific we can update this.

tapaimarton committed:
Update analysis.ini

spxiwh commented:
Using these config files I see:

```
ConfigParser.NoSectionError: No section: 'multiifo_coinc-full3d'
```

spxiwh commented:
Is the problem at line 259 of analysis.ini. Think that should be:
```
[coinc-fullinj&coinc-injfull]
; perform little-dog analysis for software injections
timeslide-interval = ${multiifo_coinc-full|timeslide-interval}
```

spxiwh commented:
With this change a workflow generates ... Seeing what happens now when I run it!

tapaimarton commented:
Ah, it seems, I missed that while copying the changes into the files to upload

tapaimarton committed:
Update analysis.ini

tdent committed:
Update analysis.ini

Edit first line to be less specific about exactly what data this is for ; simplify comment on timeslide precedence list to describe actual analysis outcome

tdent reviewed (approved):
OK from my side
